### PR TITLE
gh-87092: move CFG related code from compile.c to flowgraph.c

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-87092

### CODE REVIEW

File: Include/internal/pycore_compile.h [hunk 1]
```diff
@@ -33,6 +33,16 @@ extern int _PyAST_Optimize(
     struct _arena *arena,
     _PyASTOptimizeState *state);
 
+/* Utility for a number of growing arrays used in the compiler */
+int _PyCompile_EnsureArrayLargeEnough(
+        int idx,
+        void **arr_,
```
Comment (markshannon) [inline Include/internal/pycore_compile.h]: Maybe just `array`? I assume the odd name is a hangover from when this was macro?

File: Include/internal/pycore_opcode_utils.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+         (opcode) == STORE_FAST__LOAD_FAST || \
+         (opcode) == STORE_FAST__STORE_FAST)
+
+#define IS_SCOPE_EXIT_OPCODE(opcode) \
+        ((opcode) == RETURN_VALUE || \
+         (opcode) == RETURN_CONST || \
+         (opcode) == RAISE_VARARGS || \
+         (opcode) == RERAISE)
+
+#define IS_SUPERINSTRUCTION_OPCODE(opcode) \
```
Comment (markshannon) [inline Include/internal/pycore_opcode_utils.h]: So "super" they defined it twice 😉 
(duplicate definition)

File: Include/internal/pycore_opcode_utils.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        ((opcode) == LOAD_FAST__LOAD_FAST || \
+         (opcode) == LOAD_FAST__LOAD_CONST || \
+         (opcode) == LOAD_CONST__LOAD_FAST || \
+         (opcode) == STORE_FAST__LOAD_FAST || \
+         (opcode) == STORE_FAST__STORE_FAST)
+
+#define IS_SCOPE_EXIT_OPCODE(opcode) \
```
Comment (markshannon) [inline Include/internal/pycore_opcode_utils.h]: Duplicate definition

**Review by markshannon — CHANGES_REQUESTED** [review]
A bit of duplicated code, otherwise looks fine.

I don't think `_PyOpcode_RelativeJump` is needed any more, as all jumps are relative.

File: Include/internal/pycore_opcode_utils.h (hunk 2 above)
Comment (iritkatriel) [reply Include/internal/pycore_opcode_utils.h]: super duper

**Comment by iritkatriel** [comment]
> I don't think `_PyOpcode_RelativeJump` is needed any more, as all jumps are relative.

I agree, but I'd like to do it in a separate PR because I would need to remove usages and the diff here is large enough.

**Comment by iritkatriel** [comment]
I have made the requested changes; please review again.

**Review by markshannon — APPROVED** [review]

**Pull request merged**